nanoFramework.System.Net/Sockets/Socket.cs

```diff
@@ -65,9 +67,17 @@ public class Socket : IDisposable
         /// </remarks>
         public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType)
         {
-            m_Handle = NativeSocket.socket((int)addressFamily, (int)socketType, (int)protocolType);
```

Comment:
This is meant to be this way, in case there is an exception when creating a socket that is supposed to be returned to the caller. See the documentation [here](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.-ctor?view=netframework-4.8#System_Net_Sockets_Socket__ctor_System_Net_Sockets_AddressFamily_System_Net_Sockets_SocketType_System_Net_Sockets_ProtocolType_)
`SocketException` is supposed to be thrown and it's up the caller to deal with it.

Reply:
Understood and agreed.
SocketException is now being passed on to the caller